flaky `test_pause_while_spilling`

```
__________________________ test_pause_while_spilling ___________________________

args = (), kwds = {}

    @wraps(func)
    def inner(*args, **kwds):
        with self._recreate_cm():
>           return func(*args, **kwds)

/usr/share/miniconda3/envs/dask-distributed/lib/python3.9/contextlib.py:79: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/share/miniconda3/envs/dask-distributed/lib/python3.9/contextlib.py:79: in inner
    return func(*args, **kwds)
distributed/utils_test.py:1089: in test_func
    return _run_and_close_tornado(async_fn_outer)
distributed/utils_test.py:382: in _run_and_close_tornado
    return asyncio.run(inner_fn())
/usr/share/miniconda3/envs/dask-distributed/lib/python3.9/asyncio/runners.py:44: in run
    return loop.run_until_complete(main)
/usr/share/miniconda3/envs/dask-distributed/lib/python3.9/asyncio/base_events.py:647: in run_until_complete
    return future.result()
distributed/utils_test.py:379: in inner_fn
    return await async_fn(*args, **kwargs)
distributed/utils_test.py:1086: in async_fn_outer
    return await asyncio.wait_for(async_fn(), timeout=timeout * 2)
/usr/share/miniconda3/envs/dask-distributed/lib/python3.9/asyncio/tasks.py:479: in wait_for
    return fut.result()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    async def async_fn():
        result = None
        with dask.config.set(config):
            async with _cluster_factory() as (s, workers), _client_factory(
                s
            ) as c:
                args = [s] + workers
                if c is not None:
                    args = [c] + args
                try:
                    coro = func(*args, *outer_args, **kwargs)
                    task = asyncio.create_task(coro)
                    coro2 = asyncio.wait_for(asyncio.shield(task), timeout)
                    result = await coro2
                    validate_state(s, *workers)
    
                except asyncio.TimeoutError:
                    assert task
                    buffer = io.StringIO()
                    # This stack indicates where the coro/test is suspended
                    task.print_stack(file=buffer)
    
                    if cluster_dump_directory:
                        await dump_cluster_state(
                            s=s,
                            ws=workers,
                            output_dir=cluster_dump_directory,
                            func_name=func.__name__,
                        )
    
                    task.cancel()
                    while not task.cancelled():
                        await asyncio.sleep(0.01)
    
                    # Hopefully, the hang has been caused by inconsistent
                    # state, which should be much more meaningful than the
                    # timeout
                    validate_state(s, *workers)
    
                    # Remove as much of the traceback as possible; it's
                    # uninteresting boilerplate from utils_test and asyncio
                    # and not from the code being tested.
>                   raise asyncio.TimeoutError(
                        f"Test timeout after {timeout}s.\n"
                        "========== Test stack trace starts here ==========\n"
                        f"{buffer.getvalue()}"
                    ) from None
E                   asyncio.exceptions.TimeoutError: Test timeout after 30s.
E                   ========== Test stack trace starts here ==========
E                   Stack for <Task pending name='Task-187433' coro=<test_pause_while_spilling() running at /home/runner/work/distributed/distributed/distributed/tests/test_worker_memory.py:925> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x7f02e8ee2e50>()]>> (most recent call last):
E                     File "/home/runner/work/distributed/distributed/distributed/tests/test_worker_memory.py", line 925, in test_pause_while_spilling
E                       await asyncio.sleep(0.01)

distributed/utils_test.py:1023: TimeoutError
```

https://github.com/dask/distributed/actions/runs/3130316472/jobs/5080439841#step:18:1950
https://github.com/dask/distributed/actions/runs/3026920567/jobs/4870708288#step:18:1933

--- a/distributed/tests/test_worker_memory.py
+++ b/distributed/tests/test_worker_memory.py
@@ -953,7 +953,7 @@ async def test_pause_while_spilling(c, s, a):
             return 0
         elif a.data.fast and not a.data.slow:
             # Trigger spilling
-            return parse_bytes("1.5 GiB")
+            return parse_bytes("1.6 GiB")
         else:
             # Trigger pause, but only after we started spilling
             return parse_bytes("1.9 GiB")

--- a/distributed/tests/test_worker_memory.py
+++ b/distributed/tests/test_worker_memory.py
@@ -14,6 +14,7 @@ import pytest
 from tlz import merge
 
 import dask.config
+from dask.utils import format_bytes, parse_bytes
 
 import distributed.system
 from distributed import Client, Event, KilledWorker, Nanny, Scheduler, Worker, wait
@@ -928,11 +929,12 @@ async def test_disk_cleanup_on_terminate(c, s, a, ignore_sigterm):
 @gen_cluster(
     nthreads=[("", 1)],
     client=True,
-    worker_kwargs={"memory_limit": "10 GiB"},
+    worker_kwargs={"memory_limit": "2 GiB"},
+    # ^ must be smaller than system memory limit, otherwise that will take precedence
     config={
         "distributed.worker.memory.target": False,
-        "distributed.worker.memory.spill": 0.7,
-        "distributed.worker.memory.pause": 0.9,
+        "distributed.worker.memory.spill": 0.5,
+        "distributed.worker.memory.pause": 0.8,
         "distributed.worker.memory.monitor-interval": "10ms",
     },
 )
@@ -940,16 +942,21 @@ async def test_pause_while_spilling(c, s, a):
     N_PAUSE = 3
     N_TOTAL = 5
 
+    if a.memory_manager.memory_limit < parse_bytes("2 GiB"):
+        pytest.fail(
+            f"Set 2 GiB memory limit, got {format_bytes(a.memory_manager.memory_limit)}."
+        )
+
     def get_process_memory():
         if len(a.data) < N_PAUSE:
             # Don't trigger spilling until after all tasks have completed
             return 0
         elif a.data.fast and not a.data.slow:
             # Trigger spilling
-            return 8 * 2**30
+            return parse_bytes("1.5 GiB")
         else:
             # Trigger pause, but only after we started spilling
-            return 10 * 2**30
+            return parse_bytes("1.9 GiB")
 
     a.monitor.get_process_memory = get_process_memory
 

